src/ImpactEvaluator.sol

```diff
@@ -4,14 +4,15 @@ import "../lib/openzeppelin-contracts/contracts/access/AccessControl.sol";
 import "./Balances.sol";
 pragma solidity ^0.8.19;
 
-contract ImpactEvaluator is AccessControl, Nonces, Balances {
+contract ImpactEvaluator is AccessControl, Balances {
     uint public currentRoundIndex;
     uint public currentRoundEndBlockNumber;
     uint public currentRoundRoundReward;
 
     uint public previousRoundIndex;
     uint public previousRoundTotalScores;
     uint public previousRoundRoundReward;
```

Comment:
Do we still need `previousRoundRoundReward`?

Reply:
Yes, for calculating the payout, since `score` is a factor, and `score * previousRoundRewardsLeft` creates wrong results